src/plugins/gs-plugin-external-apps.c

```diff
@@ -457,6 +457,17 @@ force_set_app_state (GsApp *app, AsAppState state)
 	gs_app_set_state (app, state);
 }
 
+static gboolean
+refine_is_details_view (GsPluginRefineFlags flags)
+{
+	/* The details view was the only one asking for this combination of
+	 * flags so we are using this approach as a way to know if this was the
+	 * view that asked for the refine but this can change in the future */
+	return (flags & GS_PLUGIN_REFINE_FLAGS_REQUIRE_ADDONS) &&
+		(flags & GS_PLUGIN_REFINE_FLAGS_REQUIRE_ORIGIN_UI) &&
+		(flags & GS_PLUGIN_REFINE_FLAGS_REQUIRE_URL);
```

Comment:
Perhaps it would be best to introduce a new flag just for this purpose, instead of trying to make a deduction from other flags?